doc/website/blog/title-tbd.md

```diff
@@ -0,0 +1,244 @@
+service annotation. With this approach, we can implement any number of other options functions aside from `Methods`. 
+Very cool!
+
+### List: The Superior GetAll
+With optional method generation out of the way, we could return our focus to adding `GetAll`. How could we implement 
+this method in a safe fashion? Rotem suggested we base the method off of Google's API Improvment Proposal (AIP) for List, 
+[AIP-132](https://google.aip.dev/132). This approach allows a client to retrieve all entities, but breaks the retrieval 
+up into pages. As an added bonus, it also sounds better than "GetAll"!
+
+
+### List Request
+With this design, a request message would look like:
+```protobuf
+message ListUserRequest {
+  int32 page_size = 1;
+
+  string page_token = 2;
+
+  View view = 3;
+
+  enum View {
+    VIEW_UNSPECIFIED = 0;
+
+    BASIC = 1;
+
+    WITH_EDGE_IDS = 2;
+  }
+}
+```
+
+#### Page Size
+The `page_size` field allows the client to specify the maximum number of entries they want to receive in the 
+response message, subject to a maximum page size of 1000. This eliminates the issue of returning more results than the 
+client can handle in the initial `GetAll` implementation. Additionally, the maximum page size was implemented to prevent 
+a client from overburdening the server.
+
+#### Page Token
+The `page_token` field is a base64-encoded string utilized by the server to determine where the next page begins. An 
+empty token means that we want the first page.
+
+#### View
+The `view` field is used to specify whether the response should return the edge ids associated with the entities.
+
+
+### List Response
+The response message would look like:
+```protobuf
+message ListUserResponse {
+  repeated User user_list = 1;
+
+  string next_page_token = 2;
+}
+```
+
+#### List
+The `*_list` field contains page entities.
+
+#### Next Page Token
+The `next_page_token` field is a base64-encoded string that can be utilized in another List Request to retrieve the next 
+page of entities. An empty token means that this response contains the last page of entities.
+
+
+### Pagination
+With the gRPC interface determined, the challenge of implementing it began. One of the most critical design decisions 
+was how to actually implement the pagination. The naïve approach would be to use `LIMIT/OFFSET` pagination to skip over 
+the entries we've already seen. However, this approach has massive [drawbacks](https://use-the-index-luke.com/no-offset); 
+the most problematic being that the database has to _fetch all the rows it is skipping_ to get the rows we want. 
+
+#### Keyset Pagination
+A much better approach, Rotem proposed, was to use Keyset pagination. This approach is slightly more 
+complicated in that it requires the use of a unique column (or combination of columns) to order the rows, but 
+is significantly faster. This is because we can take advantage of the sorted rows to select only entries with 
```

Comment:
```suggestion
in exchange we gain a significant performance improvement. This is because we can take advantage of the sorted rows to select only entries with 
```